docs/lang/articles/contribution/dev_install.md

```diff
@@ -4,318 +4,467 @@ sidebar_position: 2
-  - Install conda and use python from conda enviroments.
+</TabItem>
 
-- If `make` fails to compile and reports
-  `fatal error: 'spdlog/XXX.h' file not found`, please try runing
-  `git submodule update --init --recursive --depth=1`.
+<TabItem value="windows">
 
-- If the build succeeded but running any Taichi code results in errors
-  like
+1. Go to [the official site](https://developer.nvidia.com/cuda-downloads) and download the installer.
+2. Choose **exe (local)** as **Installer Type**.
+3. Check if CUDA is properly installed:
 
   ```
-  Bitcode file (/tmp/taichi-tero94pl/runtime//runtime_x64.bc) not found
+  nvidia-smi
   ```
 
-  please double check `clang` is in your `PATH`:
+</TabItem>
 
-  ```bash
-  clang --version
-  # version should be >= 7
-  ```
+</Tabs>
 
-  and our **Taichi configured** `llvm-as`:
+### Install Vulkan
 
-  ```bash
-  llvm-as --version
-  # version should be >= 8
-  which llvm-as
-  # should be /usr/local/bin/llvm-as or /opt/XXX/bin/llvm-as, which is our configured installation
+You must install the Vulkan SDK in order to build Taichi with a Vulkan backend. To proceed: 
+
+<Tabs
+  defaultValue="linux"
+  values={[
+    {label: 'Linux', value: 'linux'},
+    {label: 'Windows', value: 'windows'},
+]}>
+
+<TabItem value="linux">
+
+1. Go to [Vulkan's SDK download page](https://vulkan.lunarg.com/sdk/home) and follow the instructions for your OS.
+2. Check if environment variables `VULKAN_SDK`, `PATH`, `LD_LIBRARY_PATH`, and `VK_LAYER_PATH` are updated. 
+
+  > The SDK for Ubuntu provides a `setup-env.sh` for updating these variables.
+
+3. Ensure that you have a Vulkan driver from a GPU vendor properly installed.
+
+  > On Ubuntu, check if a JSON file with a name corresponding to your GPU vendor is in: `/etc/vulkan/icd.d/` or `/usr/share/vulkan/icd.d/`.
+
+4. Check if the SDK is properly installed: `vulkaninfo`.
+
+</TabItem>
+
+<TabItem value="windows">
+
+1. Go to [Vulkan's SDK download page](https://vulkan.lunarg.com/sdk/home) and follow the instructions for your OS.
+2. Set the environment variable `VULKAN_SDK` to `C:/VulkanSDK/${YOUR_VULKAN_VERSION}`.
+3. If the SDK is properly installed, add an environment variable `TAICHI_CMAKE_ARGS` with the value `-DTI_WITH_VULKAN:BOOL=ON` to enable the Vulkan backend:
+
+  ```shell
+  $env:TAICHI_CMAKE_ARGS += " -DTI_WITH_VULKAN:BOOL=ON"
   ```
 
-  If not, please install `clang` and **build LLVM from source** with
-  instructions above in [dev_install](#installing-dependencies-1),
-  then add their path to environment variable `PATH`.
+</TabItem>
+
+</Tabs>
+
+## Troubleshooting and debugging
+
+### Gets an `llvm-as can’t be opened because Apple cannot check it for malicious software on macOS`
+
+**Workaround**
+
+- One-off: **System Preferences** **> Security & Privacy > General > Allow anyway**.
+
+- Permanent: Run `sudo spctl --master-disable` to permanently allow your MacBook to download apps from anywhere by default.
```

Comment:
Disabling the security check globally looks like a terrible idea.

Reply:
@ailzhang

Reply:
Sure let's remove the second option.